src/components/TablePersonParents.tsx

```diff
@@ -0,0 +1,58 @@
+import { Person } from '../types';
+import { PersonLink } from './PersonLink';
+
+type Props = {
+  person: Person;
+  personNames: string[];
+  findParent: (parentName: string) => Person;
+};
+
+export const TablePersonParents: React.FC<Props> = ({
+  person,
+  personNames,
+  findParent,
+}) => {
+  return (
+    <>
+      {person.motherName
+        ? (
+          <>
+            {personNames?.includes(person.motherName)
+              ? (
+                <td>
+                  <PersonLink
+                    person={
+                      findParent(person.motherName)
+                    }
+                  />
+                </td>
+              )
+              : <td>{person.motherName}</td>}
+          </>
+        )
+        : (
+          <td>-</td>
+        )}
+
```

Comment:
Soo confusing, create some variables, and check if you need so many react fragments